text/0000-style-rfcs.md

```diff
@@ -0,0 +1,323 @@
+
+
+## Style guide
+
+The [existing style guide](https://github.com/rust-lang/rust/tree/master/src/doc/style)
+will be split into two guides: one dealing with API design and similar issues
+which will be managed by the libs team, and one dealing with formatting issues
+which will be managed by the style team. Note that the formatting part of the
+guide may include guidelines which are not enforced by Rustfmt. Those are outside
+the scope of the process defined in this RFC, but still belong in that part of
+the style guide.
+
+When RFCs are accepted the style guide may need to be updated. Towards the end
+of the process, the style team should audit and edit the guide to ensure it is a
+coherent document.
+
+
+## Material goals
+
+Hopefully, the style guideline process will have limited duration, one year
+seems reasonable. After that time, style guidelines for new syntax could be
+included with regular RFCs, or the fmt-rfcs repository could be maintained in a
+less active fashion.
+
+At the end of the process, the fmt-rfcs repository should be a fairly complete
+guide for formatting Rust code, and useful as a specification for Rustfmt and
+tools with similar goals, such as IDEs. In particular, there should be a
+decision made on how configurable Rustfmt should be, and an agreed set of
+default options. The formatting style guide in the Rust repository should be a
+more human-friendly source of formatting guidelines, and should be in sync with
+the fmt-rfcs repo.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This RFC introduces more process and bureaucracy, and requires more meetings for
+some core Rust contributors. Precious time and energy will need to be devoted to
+discussions.
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+Benevolent dictator - a single person dictates style rules which will be
+followed without question by the community. This seems to work for Go, I suspect
+it will not work for Rust.
+
+Parliamentary 'democracy' - the community 'elects' a style team (via the usual
+RFC consensus process, rather than actual voting). The style team decides on
+style issues without an open process. This would be more efficient, but doesn't
+fit very well with the open ethos of the Rust community.
+
+Use the RFCs repo, rather than a new repo. This would have the benefit that
+style RFCs would get more visibility, and it is one less place to keep track of
+for Rust community members. However, it risks overwhelming the RFC repo with
+style debate.
+
+Use issues on Rustfmt. I feel that the discussions would not have enough
+visibility in this fashion, but perhaps that can be addressed by wide and
+regular announcement.
+
+Use a book format for the style repo, rather than a collection of RFCs. This
+would make it easier to see how the 'final product' style guide would look.
+However, I expect there will be many issues that are important to be aware of
+while discussing an RFC, that are not important to include in a final guide.
```

Comment:
I prefer this alternative.

I think that for this process, even more so than the Rust language, compiler, or standard libraries, the style guide is the primary product here. Being able to see the actual diff of the changes in context, especially for some of the more cross-cutting issues, would be helpful when reviewing proposed changes.

While there are issues that are important while discussing an RFC, but not important to include in the final guide, those could just be discussed in the PR itself, rather than in a separate RFC document.

A hybrid approach could work as well, in which there was a single repository for the guide and the RFCs. That way, the RFC PR could include the text of the RFC, as well as the diff of the guide.

I think that either of these approaches, a book format with RFC PRs updating the book, or a hybrid repo with both individual RFCs and the final book, would help to avoid the issues that have prompted [an RFC on the Rust RFC process about including doc changes in any new RFC](https://github.com/rust-lang/rfcs/pull/1636).